### ISSUE TITLE
[Colab환경]KoBERT pytorch sentiment analysis 예제 오류

### ISSUE THREAD

**Issue by Datamin-D** [SKTBrain/KoBERT#53]
안녕하세요, NLP 공부중인 학생입니다.

이미 아시는지 모르겠지만, ko-BERT 파이토치 예제 실행시 마지막 셀에서 문제가 발생합니다. 

라이브러리 버젼상 충돌인지 모르겠지만 어떤 부분에선지 에러가 납니다. 에러 화면은 아래와 같습니다.

감사합니다.

![koBERT오류](https://user-images.githubusercontent.com/56338789/119948570-ecfde900-bfd3-11eb-91c4-1220b5f8383f.png)

**Comment by stainblue** [comment]
안녕하세요. 저도 NLP 찍먹해보다 같은 문제를 확인했는데요
아래와 같이 `BERTClassifier` 클래스의 `forward`메소드에서 `self.bert` 호출하는 부분을 살짝 고쳤더니 어찌어찌 동작은 하는거같아요.
``` py
class BERTClassifier(nn.Module):
    def __init__(self,
                 bert,
                 hidden_size = 768,
                 num_classes=2,
                 dr_rate=None,
                 params=None):
        super(BERTClassifier, self).__init__()
        self.bert = bert
        self.dr_rate = dr_rate
                 
        self.classifier = nn.Linear(hidden_size , num_classes)
        if dr_rate:
            self.dropout = nn.Dropout(p=dr_rate)
    
    def gen_attention_mask(self, token_ids, valid_length):
        attention_mask = torch.zeros_like(token_ids)
        for i, v in enumerate(valid_length):
            attention_mask[i][:v] = 1
        return attention_mask.float()

    def forward(self, token_ids, valid_length, segment_ids):
        attention_mask = self.gen_attention_mask(token_ids, valid_length)
        

        _, pooler = self.bert(input_ids = token_ids, token_type_ids = segment_ids.long(), attention_mask = attention_mask.float().to(token_ids.device), return_dict=False)   # <-- return_dict=False 추가했습니다
        if self.dr_rate:
            out = self.dropout(pooler)
        return self.classifier(out)
```

문제 해결엔 다음 링크를 참고했어요 https://stackoverflow.com/questions/65082243/dropout-argument-input-position-1-must-be-tensor-not-str-when-using-bert

좋은 하루 되세요

**Comment by Datamin-D** [comment]
> 안녕하세요. 저도 NLP 찍먹해보다 같은 문제를 확인했는데요
> 아래와 같이 `BERTClassifier` 클래스의 `forward`메소드에서 `self.bert` 호출하는 부분을 살짝 고쳤더니 어찌어찌 동작은 하는거같아요.
> 
> ```python
> class BERTClassifier(nn.Module):
>     def __init__(self,
>                  bert,
>                  hidden_size = 768,
>                  num_classes=2,
>                  dr_rate=None,
>                  params=None):
>         super(BERTClassifier, self).__init__()
>         self.bert = bert
>         self.dr_rate = dr_rate
>                  
>         self.classifier = nn.Linear(hidden_size , num_classes)
>         if dr_rate:
>             self.dropout = nn.Dropout(p=dr_rate)
>     
>     def gen_attention_mask(self, token_ids, valid_length):
>         attention_mask = torch.zeros_like(token_ids)
>         for i, v in enumerate(valid_length):
>             attention_mask[i][:v] = 1
>         return attention_mask.float()
> 
>     def forward(self, token_ids, valid_length, segment_ids):
>         attention_mask = self.gen_attention_mask(token_ids, valid_length)
>         
> 
>         _, pooler = self.bert(input_ids = token_ids, token_type_ids = segment_ids.long(), attention_mask = attention_mask.float().to(token_ids.device), return_dict=False)   # <-- return_dict=False 추가했습니다
>         if self.dr_rate:
>             out = self.dropout(pooler)
>         return self.classifier(out)
> ```
> 
> 문제 해결엔 다음 링크를 참고했어요 https://stackoverflow.com/questions/65082243/dropout-argument-input-position-1-must-be-tensor-not-str-when-using-bert
> 
> 좋은 하루 되세요

**Comment by Datamin-D** [comment]
@stainblue 헐.. 감사합니다. 바뀐 부분 적용해서 코드  실행해볼게요! 감사합니다!!
